### ISSUE TITLE
HC-App, Enhancement: "paid" status is perpetual for paypal/credit prompt uses

### ISSUE THREAD

**Issue by diabWH3** [James-A-White/HarrierCentral#227]
Now that #170  is fixed (yay! yay! yay!), a user can pay via paypal and then self-report their payment, updating their Hash Cash record on the backend. At first, the payment prompt disappears, but if the user backs out of the app and comes back, the prompt re-appears (their "paid" status remains intact, though!).

* There are two locations for the payment prompts: one on the Upcoming Runs dashboard and the other inside an individual Run Details page
* When deciding to display the payment prompts, check against the user's "paid" status for that trail, so the notification will be consistent no matter how many times the user goes in and out of the app. It would also be consistent no matter the payment method (venmo, EU bank transfer, using personal credit, etc).
* This could also be useful when Hash Cash manually marks a user as "paid" and the user will see their app update accordingly.

**Comment by James-A-White** [comment]
Agree. Excellent suggestion. This is going to take a bit of "jiggery pokery" on the backend as payment information is usually only sent to admins. Will implement in the next few weeks.
